### PR TITLE
Fix/result shield

### PR DESCRIPTION
**Pull request by zberwaldt**
This PR has a little more than the fix for the result shield rendering early in page load.

- Installed `date-fns` as this library has a wealth of date utilities. Including time deltas, date formatting, time to duration parseing.
- Separated types for the different components. I will probably return to this for further refactoring. 
- Moved as much state to the top component and then pass down. It makes it much easier to follow the data, IMO (obviously open to suggestions and feedback).
- I do still have local state for some of the sub-components, that can probably be refactor too.
- I did consider using a context, but will revisit at a later time.
- sprinkle in spinners for data the is possibly undefined. Will likely return later to refactor, pending feedback.

### CODE REVIEW

File: components/polls/DisplayPollResults.tsx [hunk 1]
```diff
@@ -11,194 +12,181 @@ import {
...
 
-const deltaT = (date1: string | number, date2: string | number) => {
-    let deltaT_ms = null;
-    try {
-        deltaT_ms = new Date(date1).getTime() - new Date(date2).getTime()
-    } catch {
-        deltaT_ms = 1000;
-    }
-    return deltaT_ms;
+type PollResultStack = {
```
Comment (wpmonty) [inline components/polls/DisplayPollResults.tsx:41]: @tikisailor would this be relevant to the SDK you're producing?

**Review by wpmonty — COMMENTED** [review]
12k line changes to package-lock, peculiar

also, not seeing the shield at all in Vercel deploy - just shows a  "No Data" block

File: components/polls/DisplayPollResults.tsx (hunk 1 above)
Comment (0xtiki) [reply components/polls/DisplayPollResults.tsx:41]: sdk should save us from having to manually define api types. E.g. Poll and strategy in this case.

File: components/polls/DisplayPollResults.tsx [hunk 2]
```diff
@@ -11,194 +12,181 @@ import {
...
+                    </Center>
+                    <Center style={{ position: 'relative', top: '14%', paddingLeft: 'auto', paddingRight: 'auto' }}>
+                        <Text as='b' fontSize='3xl'>{totalVotes ?? (<Spinner />)}</Text>
+                    </Center>
+                    <Center style={{ position: 'relative', top: '15%', paddingLeft: 'auto', paddingRight: 'auto' }}>
+                        <Text as='b'>Unlocks in</Text>
+                    </Center>
+                    <Center style={{ position: 'relative', top: '17%', paddingLeft: 'auto', paddingRight: 'auto' }}>
+                        <Text as='b' fontSize='3xl'>
+                            { countdown && endTime ? <Timer countdown={countdown} endTime={endTime} /> : (<Spinner size="3xl" />) }
```
Comment (0xtiki) [inline components/polls/DisplayPollResults.tsx:116]: Could it be that we need to add a few ms / seconds to the timer to make sure the result is available when it runs out?

**Comment by 0xtiki** [comment]
I like the timer implementation, much cleaner with the lib. Moving state up is ok with me. I would usually keep it in the local component if it is not used elsewhere but I am fine with either way. Typing the component props is good, just be aware that a lot of the api types should come with the client/sdk so that we don't have to define them manually (Many of them are defined already but some are incomplete).

When I run local I noticed a few things. After the timer expires the result chart does not show up but instead I get the "no result" component. I think it might be due to the result data not being immediately available yet. If I come back to the page and it re-renders the results show up nicely now without the shield flicker.

I also get logged out every time I use back button from the poll result page. Probably nothing to do with this PR but maybe we could add something like a retry or a delay to the useServers() hook?

**Pull request closed without merging**